### DISCUSSION TITLE
Setting type to category

### DISCUSSION THREAD

**Discussion by MagBuchSB1** [pdpipe/pdpipe#82] [Q&A] [answered]
Hi

In many cases I need to set the data type of object-columns to "category", like so;
```python
df[<object_cols>] = df[<object_cols>].astype("category")
```
How would I go about doing this in pdpipe? I've tried different approaches using ColumnDtypeEnforcer, but to no avail...

**Comment by shaypal5** [comment, marked as answer]
OK. `ColumnDtypeEnforcer` should actually support this.

For a static column list, this works fine (at least for me), assuming `'a'` and `'b'` are the labels of you object columns:

```python
dt_enforcer = pdp. ColumnDtypeEnforcer({'a': 'category', 'b': 'category'})
post_df = dt_enforcer(df)
```

If you want the pipeline stage to convert all object columns to "category", to be determined on runtime, do this:

```python
dt_enforcer = pdp. ColumnDtypeEnforcer({pdp.cq.OfDtypes('object'): 'category'})
post_df = dt_enforcer(df)
```

This also works fine for me locally. Notice that if you use the same pipeline object several times, *and* you do not want this column list to have a fit/predict logic (meaning, to be determined once on fit time and retain this list on transform), use `fittable=False` when initializing the column qualifier object, like so:

```python
dt_enforcer = pdp. ColumnDtypeEnforcer({pdp.cq.OfDtypes('object', fittable=False): 'category'})
post_df = dt_enforcer(df)
```

That's it. Let me know if this does not answer your need.
Cheers!
